app/database/operator/server_data_operator/handlers/group.ts

```diff
@@ -114,6 +116,75 @@ const GroupHandler = (superclass: any) => class extends superclass implements Gr
 
         return records;
     };
+
+    /**
+     * handleGroupTeamsForTeam: Handler responsible for the Create/Update operations occurring on the GroupTeam table from the 'Server' schema
+     * @param {string} teamId
+     * @param {HandleGroupTeamForTeamArgs} groupMembershipsArgs
+     * @param {GroupTeam[]} groupMembershipsArgs.groupMemberships
+     * @param {boolean} groupMembershipsArgs.prepareRecordsOnly
+     * @throws DataOperatorException
+     * @returns {Promise<GroupTeamModel[]>}
+     */
+    handleGroupTeamsForTeam = async ({teamId, groups, prepareRecordsOnly = true}: HandleGroupTeamForTeamArgs): Promise<GroupTeamModel[]> => {
+        // Get existing group memberships
+        const existingGroupTeams = await queryGroupTeamsForTeam(this.database, teamId).fetch();
+
+        let records: GroupTeamModel[] = [];
+        let rawValues: GroupTeam[] = [];
+
+        // Nothing to add or remove
+        if (!groups?.length && !existingGroupTeams.length) {
+            return records;
+        }
+
+        // No groups - remove all existing ones
+        if (!groups?.length && existingGroupTeams.length) {
+            records = existingGroupTeams.map((gm) => gm.prepareDestroyPermanently());
+        }
+
+        // No existing groups - add all new ones
+        if (groups?.length && !existingGroupTeams.length) {
+            rawValues = groups.map((g) => ({id: `${g.id}-${teamId}`, team_id: teamId, group_id: g.id}));
```

Comment:
All this gain us code readability, but I think the next block (the one with the sets) already covers all these cases. 0/5 whether to keep it like this, or just keep the next block. Thoughts?

Reply:
Have changed them to if/elseif/else to match the groupMembership so it makes more sense.